src/Badge/Badge.js

```diff
@@ -0,0 +1,114 @@
+// @flow weak
+
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import { createStyleSheet } from 'jss-theme-reactor';
+import customPropTypes from '../utils/customPropTypes';
+
+const radius = 12;
+const radius2x = 2 * radius;
+
+export const styleSheet = createStyleSheet('Badge', (theme) => {
+  const { typography, palette } = theme;
+
+  return {
+    root: {
+      position: 'relative',
+      display: 'inline-block',
+      padding: `${radius2x}px ${radius2x}px ${radius}px ${radius}px`,
```

Comment:
I agree with @mbrookes. I don't think that the badge should impact the layout of the element. 
What about removing that `padding` property?